### ISSUE TITLE
[ luet build ] push/pull from several hub 

### ISSUE THREAD

**Issue by welele08** [mocaccinoOS/issue-tracker#40]
I'm starting building a luet tree upon https://github.com/mocaccinoOS/portage-tree  with my own infra. 
Those package rely on "layer/X". I have my own docker registry (and like to keep It private since my packages contain proprietary code)
I'd like to pull from either my registry and quai.io, and  pull to my private docker registry. Is there a way to do that?

**Comment by mudler** [comment]
Yes definitely! `luet build` takes `--image-repository` to define a custom image base / registry

**Comment by welele08** [comment]
Thank you for your answer.
I noticed the --image-repository or the REPO_CACHE variable in Makefile. I have a similar makefile in my repo, I change the REPO_CACHE variable to my url. It works.
My build process (a  `PACKAGES="foo/bar" make rebuild`) built dep for "layers/X", "layers/X" and then my customs packages. And it store the associated docker images in my registry. But all packages and docker images from the "portage-tree" repository are compiled/rebuilt and are not downloaded from quay.Io. 

I'd like to pull images from https://quay.io/repository/mocaccinocache/portage-amd64-cache, tag them and push them in my repo or just use with luet.  Is there a way to do that easily?

I noticed that image I build and image stored in https://quay.io/repository/mocaccinocache/portage-amd64-cache has the same tag. For exemple, `layer/gentoo-stage3-0.20201125` generate an image tagged "e1eb7be13068a1fb7956b4b68d20e74f9033956b453a2fbc7436e0dab28e50c2" . I can manually do a `docker pull quay.io/repository/mocaccinocache/portage-amd64-cache:e1eb7be13068a1fb7956b4b68d20e74f9033956b453a2fbc7436e0dab28e50c2`, then do a `docker tag` to change the prefix/url and finally do a `docker pull`. But I don't know how to get the "e1eb7be13068a1fb7956b4b68d20e74f9033956b453a2fbc7436e0dab28e50c2" tag without building the package.

**Comment by mudler** [comment]
hi @welele08, sorry to get back at you so late, I noticed your answer just now.

At the moment it's possible but it's a manual process, to retrieve the hashes you could use `luet tree images`, see also https://github.com/mocaccinoOS/portage-tree#inspect-docker-images-from-packages as an example.

I've created https://github.com/mudler/luet/issues/161 to track a separate extension for it, should be really easy to implement as `luet tree images` can output the data as JSON too, so easy to parse with other tools as well (`jq` and alike)

**Comment by welele08** [comment]
Hi,

No prob for the time! Thank you for you're answer.
I think it can be useful in case of build time dep across different repo.
